src/openzaak/components/zaken/tests/test_cloud_events.py

```diff
@@ -0,0 +1,207 @@
+
+            expected_payload = {
+                "specversion": "1.0",
+                "type": ZAAK_VERWIJDERN,
+                "source": "urn:nld:oin:00000001823288444000:zakensysteem",
+                "subject": str(zaak.uuid),
+                "dataref": reverse(zaak),
+                "datacontenttype": "application/json",
+                "data": {},
+                "time": "2025-09-23T12:00:00Z",
+            }
+
+            self.assertEqual(event_payload_copy, expected_payload)
+            self.assertIn("id", event_payload)
+
+    def test_send_zaak_geopend_cloud_event(self, mock_get_solo):
+        catalogus = CatalogusFactory.create()
+        zaaktype = ZaakTypeFactory.create(
+            uuid="4f2aa64b-eb42-491f-ba48-e27e8f66716c",
+            catalogus=catalogus,
+            concept=False,
+        )
+
+        with patch(
+            "openzaak.components.zaken.api.cloud_events.send_cloud_event.delay"
+        ) as mock_send:
+            zaaktype_url = reverse(zaaktype)
+
+            response = self.client.post(
+                reverse("zaak-list"),
+                data={
+                    "zaaktype": f"http://testserver{zaaktype_url}",
+                    "vertrouwelijkheidaanduiding": VertrouwelijkheidsAanduiding.openbaar,
+                    "bronorganisatie": "517439943",
+                    "verantwoordelijkeOrganisatie": "517439943",
+                    "registratiedatum": "2018-12-24",
+                    "startdatum": "2018-12-24",
+                    "productenOfDiensten": ["https://example.com/product/123"],
+                },
+                format="json",
+                **ZAAK_WRITE_KWARGS,
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            mock_send.assert_called_once()
+
+            args, kwargs = mock_send.call_args
+            event_payload = args[0]
+
+            zaak = Zaak.objects.get(uuid=response.data["uuid"])
+
+            event_payload_copy = dict(event_payload)
+            event_payload_copy.pop("id", None)
+
+            expected_payload = {
+                "specversion": "1.0",
+                "type": ZAAK_GEOPEND,
+                "source": "urn:nld:oin:00000001823288444000:zakensysteem",
+                "subject": str(zaak.uuid),
+                "dataref": reverse(zaak),
+                "datacontenttype": "application/json",
+                "data": {},
+                "time": "2025-09-23T12:00:00Z",
+            }
+
+            self.assertEqual(event_payload_copy, expected_payload)
+            self.assertIn("id", event_payload)
+
+    def test_send_cloud_event_function_posts_expected_payload(self, mock_get_solo):
+        service = ServiceFactory()
+
+        mock_config = mock.Mock()
+        mock_config.enabled = True
+        mock_config.source = "urn:nld:oin:00000001823288444000:zakensysteem"
+        mock_config.webhook_service = service
+        mock_config.webhook_path = "http://testserver/events"
```

Comment:
The issue is that this will try to do a post with path `http://testserver/events`, even though it expects a relative path (e.g. `/events`), but this test doesn't fail because build_client is mocked.

Instead you should create service and use `requests_mock` to mock the expected URL